fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/LinkedOrganizationsHandler.kt

```diff
@@ -24,4 +31,84 @@ constructor(
   fun handleChallenge(context: HandlerContext, challenge: Challenge) {
     linkedOrgRepo.updateChallenge(challenge)
   }
+
+  /**
+   * Process a Federation Result message
+   *
+   * @param context the HandlerContext of the message
+   * @param result the message that was received
+   */
+  @Throws(UnknownLaoException::class)
+  fun handleResult(context: HandlerContext, result: FederationResult) {
+    if (result.isSuccess()) {
+      if (result.challenge.data == linkedOrgRepo.getChallenge() &&
+          linkedOrgRepo.otherLaoId != null) {
+        val laoId = context.channel.extractLaoId()
+        linkedOrgRepo.addLinkedLao(laoId, linkedOrgRepo.otherLaoId!!, arrayOf())
+        laoRepo.addDisposable(
+            context.messageSender
```

Comment:
Maybe for better separability it would be nice to have an ad-hoc disposable in the linked org repo, without using the lao one

Reply:
Yes, I'll do that